packages/cli/internal/pkg/cli/logs_engine.go

```diff
@@ -45,10 +72,29 @@ func (o *logsEngineOpts) Execute() error {
 	}
 
 	logGroupName := contextInfo.EngineLogGroupName
-	if o.tail {
-		err = o.followLogGroup(logGroupName)
+	log.Debug().Msgf("Engine log group name: '%s'", logGroupName)
+
+	if o.workflowRunId != "" {
```

Comment:
This method is getting a bit too big. Could you please simplify it by extracting independent flows into separate methods. I would prefer it to look like this:

```go
if o.workflowRunId != "" {
  return o.executeGetWholeEngineLog()
} else {
  return o.executeGetEngineLogForRunId()
}
```